tests/Integ/S3Context.php

```diff
@@ -280,4 +145,120 @@ public function iChangeTheBodyOfThePreSignedRequestToBe($body)
         $this->presignedRequest = $this->presignedRequest
             ->withBody(Psr7\stream_for($body));
     }
+
+
+    /**
+     * @Given I have an s3 client and I have a file
+     */
+    public function iHaveAnClientAndIHaveAFile()
+    {
+        $this->s3Client = self::getSdk()->createS3();
+        $this->stream = Psr7\stream_for(Psr7\try_fopen(self::$tempFile, 'r'));
+    }
+
+    /**
+     * @When I create a POST object with policy condition :version:
+     */
+    public function iCreateAPostObjectWithPolicyCondition(
+        $version,
+        TableNode $table
+    ){
+        date_default_timezone_set('UTC');
+
+        $policy = [
+            'expiration' => gmdate('Y-m-d\TH:i:s\Z', strtotime('+1 hours')),
+            'conditions' => [
+                ["bucket" => self::getResourceName()],
+                ["starts-with", '$key', ""],
+            ],
+        ];
+
+        $args = [];
+        foreach ($table as $row) {
+            $args[$row['key']] = $row['value'];
+        }
+
+        if ($version == 'v2') {
```

Comment:
Rather than branching here, why not just create two different methods? V2 and V4 signing have different interfaces.